eos-paygd/eos-paygd.service.in

```diff
@@ -1,14 +1,22 @@
 [Unit]
 Description=Pay As You Go Daemon
 Documentation=man:eos-paygd(8)
-Before=systemd-user-sessions.service
+DefaultDependencies=no
+# Only run from initrd
+ConditionPathExists=/etc/initrd-release
+# Don't complete boot without this running
+OnFailure=emergency.target
+# Start early to reduce the chance the user is dropped to a shell before we've started
```

Comment:
According to the symlink created in the install script, this service is activated when initrd-root-fs.target enters the startup queue. While the followup conditions you have here will make it execute fairly early, maybe you should make it wanted by basic.target for a bit more clarity.

With or without that change, your service here will potentially start before or during the fsck of the root device. That may take a few minutes. But I guess that's accounted for by the 20 minute wait for the system dbus daemon to appear.

Reply:
Would the following make sense:
```
[Unit]
...
After=sysinit.target
Before=basic.target
...
[Install]
WantedBy=basic.target
```
Though we don't need an `[Install]` section because dracut is installing it, I guess you're saying do it as documentation?

I think doing it any earlier than `sysinit.target` seems risky in case something is not yet available that's needed (like a tmpfile).

Reply:
As currently written we wait 10 minutes for the primary root filesystem, then 20 minutes for the D-Bus daemon. But I think surely that's enough time for fsck to happen.

Reply:
I think the best way to increase consistency with the comment that you want this to run early, is to modify module-setup.sh to make this wanted by basic.target.

Reply:
Done